core/src/main/java/org/elasticsearch/index/mapper/internal/SeqNoFieldMapper.java

```diff
@@ -108,6 +114,34 @@ public Query termQuery(Object value, @Nullable QueryShardContext context) {
             throw new QueryShardException(context, "SeqNoField field [" + name() + "] is not searchable");
         }
 
+        @Override
+        public FieldStats stats(IndexReader reader) throws IOException {
+            final List<LeafReaderContext> leaves = reader.leaves();
+            if (leaves.isEmpty()) {
+                return null;
+            }
+
+            long currentMin = Long.MAX_VALUE;
+            long currentMax = Long.MIN_VALUE;
+            boolean found = false;
+            for (int i = 0; i < leaves.size(); i++) {
+                final LeafReader leaf = leaves.get(i).reader();
+                final NumericDocValues values = leaf.getNumericDocValues(name());
+                if (values == null) continue;
+                found = true;
+                final Bits bits = leaf.getLiveDocs();
+                for (int docID = 0; docID < leaf.maxDoc(); docID++) {
+                    if (bits == null || bits.get(docID)) {
+                        final long value = values.get(docID);
+                        currentMin = Math.min(currentMin, value);
+                        currentMax = Math.max(currentMax, value);
+                    }
+                }
```

Comment:
I cant think of any place where we would want a full top level scan like this. Can we return -1 or null or throw uoe? This acts O(n^2) in a near real time system. Do we know of any other loops like this in this stats code?